pkg/cosign/keys.go

```diff
@@ -51,6 +60,73 @@ func GeneratePrivateKey() (*ecdsa.PrivateKey, error) {
 	return ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
 }
 
+func ImportKeyPair(keyPath string, pf PassFunc) (*Keys, error) {
+
+	kb, err := os.ReadFile(filepath.Clean(keyPath))
+	if err != nil {
+		return nil, err
+	}
+
+	p, _ := pem.Decode(kb)
+	if p == nil {
+		return nil, errors.New("invalid pem block")
+	}
+
+	if p.Type == RSAPrivateKeyPemType {
+		pk, err := x509.ParsePKCS1PrivateKey(p.Bytes)
+		if err != nil {
+			return nil, fmt.Errorf("parsing error")
+		}
+		return MarshallKeyPair(Key{pk, pk.Public()}, pf)
+	} else if p.Type == ECPrivateKeyPemType {
+
+		pk, err := x509.ParseECPrivateKey(p.Bytes)
+		if err != nil {
+			return nil, fmt.Errorf("parsing error")
+		}
+		return MarshallKeyPair(Key{pk, pk.Public()}, pf)
+	} else {
+		return nil, fmt.Errorf("unsupported pem type: %s", p.Type)
+	}
+
+}
+
+func MarshallKeyPair(keypair Key, pf PassFunc) (*Keys, error) {
```

Comment:
nit: spelling `MarshalKeyPair`

Reply:
could you share this code with `GenerateKeyPair`? and given that it's only called within this function, maybe it should be made private.

Reply:
made changes.